apis/controlplane/v1alpha4/nestedcontrolplane_types.go

```diff
@@ -17,29 +17,83 @@ limitations under the License.
 package v1alpha4
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // NestedControlPlaneSpec defines the desired state of NestedControlPlane
 type NestedControlPlaneSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// EtcdRef is the reference to the NestedEtcd
+	EtcdRef *corev1.ObjectReference `json:"etcd,omitempty"`
+
+	// APIServerRef is the reference to the NestedAPIServer
+	// +optional
+	APIServerRef *corev1.ObjectReference `json:"apiserver,omitempty"`
 
-	// Foo is an example field of NestedControlPlane. Edit NestedControlPlane_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// ContollerManagerRef is the reference to the NestedControllerManager
+	// +optional
+	ControllerManagerRef *corev1.ObjectReference `json:"controllerManager,omitempty"`
 }
 
 // NestedControlPlaneStatus defines the observed state of NestedControlPlane
 type NestedControlPlaneStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Etcd stores the connection information from the downstream etcd
+	// implementation if the NestedEtcd type isn't used this
+	// allows other component controllers to fetch the endpoints.
+	// +optional
+	Etcd *NestedControlPlaneStatusEtcd `json:"etcd,omitempty"`
+
+	// APIServer stores the connection information from the control plane
+	// this should contain anything shared between control plane components
+	// +optional
+	APIServer *NestedControlPlaneStatusAPIServer `json:"apiserver,omitempty"`
+
+	// Initialized denotes whether or not the control plane has the
```

Comment:
Is the comment valid? What is the config-map?

Reply:
Good catch, this is a KubeAdm thing where the config is uploaded to a configmap in the kube-system namespace. This might be relevant later but for now you are right and I'll update the comment.